src/Currencies.jl

```diff
@@ -23,10 +23,15 @@ export Currency
 """
 This is a singleton type, intended to be used as a label for dispatch purposes
 """
-struct Currency{S} end
+struct Currency{S}
+    function Currency(S::Symbol)
```

Comment:
This should simply be: `Currency(S::Symbol) = (_currency_data[S][1])()`.  That will give an error if S is not found.